server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -119,6 +122,124 @@ const orderMutations = {
       return models.Order.findOne(query);
     },
   },
+  updateOrder: {
+    type: Order,
+    description: "Update an Order's amount, tier, payment method, or frequency",
+    args: {
+      order: {
+        type: new GraphQLNonNull(OrderReferenceInput),
+        description: 'Object matching the OrderReferenceInput (id)',
+      },
+      amount: {
+        type: GraphQLInt,
+        description: 'Amount in cents of the order',
+      },
+      frequency: {
+        type: GraphQLString,
+        description: 'Frequency of the recurring order, either MONTHLY or YEARLY',
+      },
+      tier: {
+        type: GraphQLString,
+        description: 'The tier of the recurring contribution, like Backer or Sponsor',
+      },
+      paymentMethod: {
+        type: PaymentMethodReferenceInput,
+        description: 'Object matching the PaymentMethodReferenceInput (uuid)',
+      },
+    },
+    async resolve(_, args, req) {
+      const decodedId = getDecodedId(args.order.id);
+
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to update a subscription');
+      }
+
+      const { paymentMethod } = args;
+
+      const query = {
+        where: {
+          id: decodedId,
+        },
+        include: [{ model: models.Subscription }, { model: models.PaymentMethod, as: 'paymentMethod' }],
+      };
+
+      let order = await models.Order.findOne(query);
+
+      if (!order) {
+        throw new NotFound('Subscription not found');
+      }
+      if (!req.remoteUser.isAdmin(order.FromCollectiveId)) {
+        throw new Unauthorized("You don't have permission to update this subscription");
+      }
+      if (!order.Subscription.isActive) {
+        throw new Error('Subscription must be active to be updated');
+      }
+
+      // payment method
+      if (paymentMethod !== undefined) {
+        // unlike v1 we don't have to check/assign new payment method, that will be taken care of in another mutation
+        const newPaymentMethod = await models.PaymentMethod.findOne({
+          where: { uuid: paymentMethod.uuid },
+        });
+        if (!newPaymentMethod) {
+          throw new Error('Payment method not found with this uuid', paymentMethod.uuid);
+        }
+        if (!req.remoteUser.isAdmin(paymentMethod.CollectiveId)) {
```

Comment:
Should be newPaymentMethod not paymentMethod.

Reply:
I get that `CollectiveId` could be existing in current version but that's not to be trusted.

Reply:
What parameter would you recommend using to check that the user is allowed to use the payment method?